### PR TITLE
Fixing Bugs in Cron Script

### LINKED ISSUE #363: Cron Script for Installing Security Updates is failing for AWS and Azure
Unfortunately the cron script for the security updates is now failing for AWS and Azure. Investigation is ongoing. Will follow up in this thread. For the time being the following actions have been taken:

- Image v050 has been rebuilt from master
- A new pre-release v051b1 has been created to be used as test
- cron script has been disabled

### PR DESCRIPTION
**Pull request by carmat88**
## Change content and motivation
This PR is meant to fix found bugs in the cron script. See changes for further comments.

Fixes: https://github.com/kubenow/KubeNow/issues/363

### CODE REVIEW

File: bin/clean_az_dupl.sh [hunk 1]
```diff
@@ -13,8 +13,8 @@ az login --service-principal -u "$AZURE_CLIENT_ID" -p "$AZURE_CLIENT_SECRET" --t
 echo -e "-----------  -----------  ---------------------------  -------  ------------------------------------\n"
 echo -e "Check if $IMAGE_NAME already exists:\n"
 
-# Extracting id of the last successful built artificat. Useful later when we need to
-# evaluate if there are any duplicates. If so, then all namesake ones with a different id will be annihilated
+# Extracting id of the last successful built artifact.
+# Useful later when we need to evaluate if there are any duplicates.
```
Comment (carmat88) [inline bin/clean_az_dupl.sh:17]: Just shortening comments for an easier reading and shorter scripts

File: bin/clean_os_dupl.sh [hunk 2]
```diff
@@ -9,8 +9,8 @@ sudo pip install --upgrade python-glanceclient
 echo -e "OPENSTACK\n---------"
 echo -e "Check if $IMAGE_NAME already exists:\n"
 
-# Extracting id of the last successful built artificat. Useful later when we need to
-# evaluate if there are any duplicates. If so, then all namesake ones with a different id will be annihilated
+# Extracting id of the last successful built artifact.
+# Useful later when we need to evaluate if there are any duplicates.
```
Comment (carmat88) [inline bin/clean_os_dupl.sh:13]: Same here, shortening comments for an easier reading and shorter scripts

File: bin/cron_script.sh [hunk 3]
```diff
@@ -20,8 +20,11 @@ if [ "$HOST_CLOUD" = 'aws' ]; then
   # Installing aws-cli
   pip install awscli --upgrade
 
+  # Getting aws owner-id. Useful in order to avoid receiving public or private results
+  # Also, easier to perform migration to new aws accounts
+  aws_owner_id=$(aws sts get-caller-identity --output text --query 'Account')
   # Extracting ami ID of latest kubenow stable (usually in the format of "ImageId:" "ami-xxxxxxx",)
-  kubenow_latest_amiId=$(aws ec2 describe-images --filters "Name=name,Values=kubenow-$CURRENT_VERSION" "Name=owner-id,Values=105135433346" |
+  kubenow_latest_amiId=$(aws ec2 describe-images --filters "Name=name,Values=kubenow-$CURRENT_VERSION" "Name=owner-id,Values=$aws_owner_id" |
```
Comment (carmat88) [inline bin/cron_script.sh:27]: As explained on lines 23-25, before we were using an hard-coded owner id value. Forgot to change this value in the previous PR about aws refractory. So this was the main root within the cron-script bug (for AWS). However, a more convenient and correct way is to NOT have it hard-coded. Thus, now is being queried from Amazon and saved in a local variable. Also good for future migration.

File: bin/cron_script.sh [hunk 4]
```diff
@@ -52,6 +55,7 @@ elif [ "$HOST_CLOUD" = 'azure' ]; then
 
   # Finally, inserting image_url attribute in build-azure.json
   sed -i -e '/os_type/a \         "image_url": "'"$kn_vhd_url"'", ' build-azure.json
+  echo -e "Kubenow latest stable VHD url is: $kn_vhd_url .\n"
```
Comment (carmat88) [inline bin/cron_script.sh:58]: Added in order to have at least one feedback output line about changes performed by the cron_script when it comes down to Azure. Azure was the other provider failing, but I believe not for any bug reasons. Seems to me it was a one-off error.

File: bin/cron_script.sh [hunk 5]
```diff
@@ -62,6 +66,6 @@ else
   echo -e "HOST_CLOUD is NOT set to one of the following values: aws, azure, gce, openstack .\n"
 fi
 
-# Last but not least, replacing the script attribute in the provisioner section to a different value than default one (i.e. requirements.sh)
-# Given that we are starting from a stable kubenow image, we do not want to reinstall the requirements, rather to perform security updates
+# Finally, overwriting provisioner's default script attribute (i.e. requirements.sh)
+# Reason: using a stable image, no need to reinstall all requirements, only security updates
```
Comment (carmat88) [inline bin/cron_script.sh:70]: Shortening comments for an easier reading and shorter scripts

File: build-azure.json [hunk 6]
```diff
@@ -16,7 +16,7 @@
          "tenant_id": "{{user `arm_tenant_id`}}",
 
          "location": "West Europe",
-         "vm_size": "Standard_A1",
+         "vm_size": "Standard_D2_v2",
```
Comment (carmat88) [inline build-azure.json:19]: Main reason behind this upgrade is about saving some build time (up to 8 minutes less per build). Not sure in terms of costs, but in principle we should be even out as yielding shorter build process, despite using a more powerful machine.

**Comment by carmat88** [comment]
@mcapuccini Hey Marco. I've managed to fix all errors. Please whenever you have some time, have a look and let me know what you think. Once merged with the master, I will also apply changes into the stable branch and set up the script so to work for a week or so only with the "pre-release" 051b1. Thanks a lot. Cheers.

**Review by mcapuccini — APPROVED** [review]
LGTM!

**Pull request merged**